rir/src/compiler/pir/instruction.h

```diff
@@ -412,25 +412,42 @@ class Instruction : public Value {
     PirType inferredTypeForArithmeticInstruction(const GetType& getType) const {
         auto m = mergedInputType(getType);
         if (!m.maybeObj()) {
-            auto t = PirType::bottom();
+            // Everything but numbers throws an error
+            auto t = m & PirType::num().notMissing().orAttribsOrObj();
```

Comment:
why was it neccessary to add `m` here?

It seems the old strategy was start add bottom, add what is needed. The new strategy seems start with number and remove what is not needed. the old strategy seems to pontentially be more precise, not?

Reply:
good point